### PR TITLE
fix show all feature for json tree

### PR DESCRIPTION
**Pull request by steppy452**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #245

## What is the new behavior?
* `showAll` defaults infinity list size to `Infinity` and ignores showAllLimit. it also gets propagated through to child nodes
* `showAllLimit` gets propagated through to child nodes. also updated the comment to show when `showAll` is false
* added storybook story to demonstrate `showAll`

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/layout/Tree/JsonTree/JsonTree.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export interface JsonTreeProps {
   showAll?: boolean;
 
   /**
-   * The limit for the number of nodes to show when `showAll` is true.
+   * The limit for the number of nodes to show when `showAll` is false.
    */
   showAllLimit?: number;
```
Comment (ghsteff) [inline src/layout/Tree/JsonTree/JsonTree.tsx:20]: `showAllLimit` is kind of a confusing name imo, but I get you're not tryna break the API. Not even sure what a good name would be anyway, maybe `expandCount` - similar to the existing `expandDepth`

**Review by ghsteff — APPROVED** [review]

File: src/layout/Tree/JsonTree/JsonTree.tsx (hunk 1 above)
Comment (ghsteff) [reply src/layout/Tree/JsonTree/JsonTree.tsx:20]: or `expandLimit`

Comment (ghsteff) [reply src/layout/Tree/JsonTree/JsonTree.tsx:20]: eh none of those are great haha

Comment (amcdnl) [reply src/layout/Tree/JsonTree/JsonTree.tsx:20]: Agreed odd name but she just followed the other name. I think fine for now.

**Review by amcdnl — APPROVED** [review]

File: src/layout/Tree/JsonTree/JsonTree.tsx (hunk 1 above)
Comment (steppy452) [reply src/layout/Tree/JsonTree/JsonTree.tsx:20]: yea didn't want to change it as it'd be a breaking change

**Pull request merged**